### DISCUSSION TITLE
Presenting Typst Slides

### DISCUSSION THREAD

**Discussion by pacaunt** [stiglers-eponym/BeamerPresenter#35] [Q&A] [answered]
How I can make beamerpresenter know the overlays of the Typst pdf?

**Comment by stiglers-eponym** [comment]
BeamerPresenter detects overlays by the page labels of the PDF. These page labels are the page numbers shown in a PDF viewer (if it does not show only the page index). In Typst, it seems like the PDF does not contain any page labels. Thus, BeamerPresenter simply does not know what are the overlays.

**Comment by stiglers-eponym** [comment]
When using Typst to create slides, do you use [Polylux](https://polylux.dev/book)? Polylux provides the tool [polylux2pdfpc](https://polylux.dev/book/external/pdfpc.html) for [pdfpc](https://pdfpc.github.io/), which writes some information including overlay information in a JSON file. If you want to use that in BeamerPresenter, I could try to add an option to read overlays from such a JSON file. But I would need a sample JSON file (or better PDF + JSON).

**Comment by stiglers-eponym** [comment, marked as answer]
I implemented reading the JSON file created by Polylux as experimental feature in the `dev` branch. Using that requries the following steps:
1. Use [polylux2pdfpc](https://polylux.dev/book/external/pdfpc.html#extracting-the-data--polylux2pdfpc) to create a .pdfpc file (this is just a JSON file) containing the overlay information.
2. Go to "Settings -> rendering -> select pdfpc/JSON file ..." and select the .pdfpc file in the appearing file dialog.

Please note: This feature is not yet available in the main branch.

**Reply by stiglers-eponym** [reply]
Update: this feature is included in version 0.2.6
